src/lib/components/Communities/CommunityHeader.js

```diff
@@ -0,0 +1,92 @@
+// This file is part of InvenioRDM
+// Copyright (C) 2020-2021 CERN.
+// Copyright (C) 2020-2021 Northwestern University.
+// Copyright (C) 2021 Graz University of Technology.
+//
+// Invenio RDM Records is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+
+import React from 'react';
+import { Container, Image, Icon } from 'semantic-ui-react';
+import PropTypes from 'prop-types';
+
+class CommunityHeaderIcon extends React.Component {
+  render() {
+    return (
+      <Icon
+        className="ml-5"
+        link
+        color="grey"
+        onClick={() => console.log('TODO: Open community modal')}
+        name="setting"
+      />
+    );
+  }
+}
+
+class CommunityHeaderInfo extends React.Component {
+  render() {
+    const { community, imagePlaceholderLink } = this.props;
+    const logoLink = community ? community?.links?.logo : imagePlaceholderLink;
+    return (
+      <Container className="community-header">
+        <Image size="mini" src={logoLink} /> {community?.metadata?.title}
+        <CommunityHeaderIcon />
+      </Container>
+    );
+  }
+}
+
+CommunityHeaderInfo.propTypes = {
+  imagePlaceholderLink: PropTypes.string.isRequired,
+  community: PropTypes.object,
+};
+
+CommunityHeaderInfo.defaultProps = {
+  community: undefined,
+};
+
+class EmptyCommunity extends React.Component {
+  render() {
+    return (
+      <Container className="community-header empty">
+        <span>No community selected.</span>
+        <CommunityHeaderIcon />
+      </Container>
+    );
+  }
+}
+
+export class CommunityHeaderComponent extends React.Component {
+  componentDidMount() {
+    const { community, setCommunity } = this.props;
+    setCommunity(community);
+  }
+
+  render() {
+    const { communityRedux, imagePlaceholderLink } = this.props;
```

Comment:
sorry naming police. I would say component should not know about redux even from names, why do we have `community` and `communityRedux`? could it be one prop ? if there is this separation maybe a parent should pass community to this component
```suggestion
    const { communityRedux, imagePlaceholderLink } = this.props;
```

Reply:
The `community` is the initial value and the `communityRedux` is the value stored in the client, I kept both as we will need to compare the values to check if they changed and do different actions in consequence (if I am not mistaken). 
We could probably rename it to `initialCommunity` and might be needed to be stored in redux too.

Reply:
why do we need to compare the initial community and the one in redux ? 
I am probably missing some context of how this app works, but why not to store the initial community in redux when the whole app is initalized first and then manage the state and changes from redux?

Reply:
the reason of this implementation was explained in a call with @zzacharo